vunk-lexer/src/lib.rs

```diff
@@ -80,129 +78,160 @@ impl std::fmt::Display for Token {
-        let op_div = just('/').map(|c| Token::Op(c.to_string()));
-        let op_rem = just('%').map(|c| Token::Op(c.to_string()));
-        let op_eq = just("==").map(|c| Token::Op(c.to_string()));
-        let op_neq = just("!=").map(|c| Token::Op(c.to_string()));
-        let op_less = just('<').map(|c| Token::Op(c.to_string()));
-        let op_less_eq = just("<=").map(|c| Token::Op(c.to_string()));
-        let op_more = just('>').map(|c| Token::Op(c.to_string()));
-        let op_more_eq = just(">=").map(|c| Token::Op(c.to_string()));
-
-        let op_bit_and = just('&').map(|c| Token::Op(c.to_string()));
-        let op_logical_and = just("&&").map(|c| Token::Op(c.to_string()));
-        let op_bit_or = just('|').map(|c| Token::Op(c.to_string()));
-        let op_logical_or = just("||").map(|c| Token::Op(c.to_string()));
-
-        let op_bit_xor = just('^').map(|c| Token::Op(c.to_string()));
-
-        let op_join = just("++").map(|c| Token::Op(c.to_string()));
-
-        op_add
-            .or(op_sub)
-            .or(op_mul)
-            .or(op_div)
-            .or(op_rem)
-            .or(op_eq)
-            .or(op_neq)
-            .or(op_less)
-            .or(op_less_eq)
-            .or(op_more)
-            .or(op_more_eq)
-            .or(op_bit_and)
-            .or(op_logical_and)
-            .or(op_bit_or)
-            .or(op_logical_or)
-            .or(op_bit_xor)
-            .or(op_join)
+        // Put the "long" tokens first, so they get parsed first
+        chumsky::primitive::choice((
+            chumsky::primitive::choice((
+                just("->").map(Token::Op),
+                just("==").map(Token::Op),
+                just("!=").map(Token::Op),
+                just("<=").map(Token::Op),
+                just(">=").map(Token::Op),
+                just("&&").map(Token::Op),
+                just("||").map(Token::Op),
+                just("++").map(Token::Op),
+            )),
+            chumsky::primitive::choice((
+                just("<").map(Token::Op),
+                just(">").map(Token::Op),
+                just("&").map(Token::Op),
+                just("|").map(Token::Op),
+                just("(").map(Token::Op),
+                just(")").map(Token::Op),
+                just(",").map(Token::Op),
+                just("=").map(Token::Op),
+                just(":").map(Token::Op),
+                just(".").map(Token::Op),
+                just(";").map(Token::Op),
+                just("[").map(Token::Op),
+                just("]").map(Token::Op),
+                just("{").map(Token::Op),
+                just("}").map(Token::Op),
+                just("|").map(Token::Op),
+                just("+").map(Token::Op),
+                just("-").then_ignore(just(">").not()).map(Token::Op),
+                just("*").map(Token::Op),
+                just("/").map(Token::Op),
+                just("%").map(Token::Op),
+                just("^").map(Token::Op),
+            )),
```

Comment:
Because all of these `just`s have the same output type (`&str`) you can move the `.map(Token::Op)` to after the `choice`, if you prefer.